llvm/lib/CheerpUtils/TypeOptimizer.cpp

```diff
@@ -289,6 +289,8 @@ bool TypeOptimizer::isUnsafeDowncastSource(StructType* st)
 
 bool TypeOptimizer::canCollapseStruct(llvm::StructType* st, llvm::StructType* newStruct, llvm::Type* newType)
 {
+	if (newStruct == nullptr)
```

Comment:
add a comment that explains that this happens if st is a literal. 
Even better, just assert that st->isLiteral before returning false

Reply:
I've put the assertion there, good call